js/bybit.js

```diff
@@ -3215,7 +3579,11 @@ module.exports = class bybit extends Exchange {
          * @returns {object} An [order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
          */
         if (symbol === undefined) {
-            throw new ArgumentsRequired (this.id + ' cancelOrder() requires a symbol argument');
+            let type = undefined;
+            [ type, params ] = this.handleMarketTypeAndParams ('cancelOrder', undefined, params);
+            if (type !== 'spot') {
+                throw new ArgumentsRequired (this.id + ' cancelOrder() requires a symbol argument');
+            }
         }
         await this.loadMarkets ();
         const market = this.market (symbol);
```

Comment:
@ttodua this logic isn't right, if you don't provide any symbol in the spot mode, will crash here

If symbol is not required for spot orders, so we can only `this.market (symbol);` if the users provide a symbol

Reply:
Ah, good catch. the logic needed deeper edits. I've updated, and take a look, if it seems 'sketchy' let me know. (however I dont see other way, if we want to maintain that flexibility so user was able to skip the symbol )